go-controller/pkg/cluster/master.go

```diff
@@ -400,3 +405,7 @@ func (cluster *OvnClusterController) watchNodes() error {
 	}, cluster.syncNodes)
 	return err
 }
+
+func (cluster *OvnClusterController) stopWatchingNodes() {
+	cluster.watchFactory.RemoveHandlers()
```

Comment:
@numansiddique I think this will kill all handlers for the factory. Instead we want to save the handler ID of our node watch into OvnClusterController and the RemoveHandler(<id>) when we want to stop it.

Also, I think this function can be collapsed into StopClusterMaster()? That's the only caller that I can see.

Reply:
When a ovnkube-master transitions from leader to a follower, I thought it should not watch on any events from apiserver and should kill all handlers. 
That's why I thought of doing this way.

Is there a need for the ovnkube follower to still monitor for events from api-server ?

Reply:
As discussed offline, its better for now for ovnkube master to exit when it transitions from leader to follower.

Reply:
As a follow-up, we decided to have the master just exit when it loses election, because there is more to do than just removing watch events to make things work (like clearing caches in the OvnController).

Reply:
Done